bpf/tc_http_tp.c

```diff
@@ -347,34 +347,20 @@ static __always_inline int is_http_request(struct __sk_buff *ctx) {
         return 0;
     }
 
-    char req_buf[] = "OPTIONS /"; // largest HTTP request operation
+    unsigned char req_buf[] = "OPTIONS /"; // largest HTTP request operation
 
-    const __u32 offset = (void *)payload - ctx_data(ctx);
+    const u32 offset = (void *)payload - ctx_data(ctx);
 
     if (bpf_skb_load_bytes(ctx, offset, &req_buf, sizeof(req_buf) - 1) != 0) {
         return 0;
     }
 
-    return (req_buf[0] == 'G' && req_buf[1] == 'E' && req_buf[2] == 'T' && req_buf[3] == ' ' &&
-            req_buf[4] == '/') ||
-           (req_buf[0] == 'P' && req_buf[1] == 'O' && req_buf[2] == 'S' && req_buf[3] == 'T' &&
-            req_buf[4] == ' ' && req_buf[5] == '/') ||
-           (req_buf[0] == 'P' && req_buf[1] == 'U' && req_buf[2] == 'T' && req_buf[3] == ' ' &&
-            req_buf[4] == '/') ||
-           (req_buf[0] == 'P' && req_buf[1] == 'A' && req_buf[2] == 'T' && req_buf[3] == 'C' &&
-            req_buf[4] == 'H' && req_buf[5] == ' ' && req_buf[5] == '/') ||
-           (req_buf[0] == 'D' && req_buf[1] == 'E' && req_buf[2] == 'L' && req_buf[3] == 'E' &&
-            req_buf[4] == 'T' && req_buf[5] == 'E' && req_buf[6] == ' ' && req_buf[7] == '/') ||
-           (req_buf[0] == 'H' && req_buf[1] == 'E' && req_buf[2] == 'A' && req_buf[3] == 'D' &&
-            req_buf[4] == ' ' && req_buf[5] == '/') ||
-           (req_buf[0] == 'O' && req_buf[1] == 'P' && req_buf[1] == 'T' && req_buf[1] == 'I' &&
-            req_buf[1] == 'O' && req_buf[1] == 'N' && req_buf[1] == 'S' && req_buf[1] == ' ' &&
-            req_buf[1] == '/');
+    return is_http_request_buf(req_buf, 9);
```

Comment:
```suggestion
    return is_http_request_buf(req_buf, sizeof(req_buf) - 1);
```

rationale: avoid magic constant - it's clearer and less error prone